src/utils/calculator.ts

```diff
@@ -0,0 +1,37 @@
+import { GatewayOracle } from 'ardrive-core-js';
+import regression, { DataPoint } from 'regression';
+
+export class Calculator {
+	private static readonly sampleByteVolumes = [
+		100000, // 100KB
+		100000000, // 100MB
+		10000000000 // 10GB
+	];
+	private sampleWinstonValues: number[] = [];
+	private regressionInstance?: regression.Result;
+
+	public async setup(): Promise<void> {
+		await this.fetchData();
+		this.runTheLinearRegression();
+	}
+
+	private async fetchData(): Promise<void> {
+		const oracle = new GatewayOracle();
+		const winstonSamplesQuery = Calculator.sampleByteVolumes.map((sampleByteCount) =>
+			oracle.getWinstonPriceForByteCount(sampleByteCount)
+		);
+		const winstonSamplesResult = await Promise.all(winstonSamplesQuery);
+		this.sampleWinstonValues = winstonSamplesResult;
+	}
+
+	private runTheLinearRegression(): void {
```

Comment:
Could you explain why you're using linear regression here?

Reply:
Oh you asked this in the meeting.
So, we are using linear regression to avoid querying the nodes each time the user changes the value. We instead request three values, and then predict the estimation by using that regression model